lib/mapCache.js

```diff
@@ -0,0 +1,16 @@
+'use strict'
+
+module.exports = function createMapCache (fetch, fallback) {
+  const cache = new Map()
+
+  return function get (id) {
+    if (id) {
+      if (!cache.has(id)) {
+        cache.set(id, fetch.apply(null, arguments))
+      }
+      return cache.get(id)
+    } else {
+      return (fallback || fetch).apply(null, arguments)

```

Comment:
As a safety check, this is only used to bootstrap the server, right? In case, can you add it as a comment at the top? `.apply(null, arguments)` is __extremely slow__ on Node 4 and 6.
A better approach should be to convert it to an array:

```js
var args = new Array(arguments.length)
for (var i = 0; i < args.length; i++) { args[i] = arguments[i] }
```

Reply:
Yes, this should only be used to bootstrap the server - I've added a comment and added the conversion to an array before calling.